src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/VisitSchedulerService.kt

```diff
@@ -200,4 +213,37 @@ class VisitSchedulerService(
   fun getNotificationsTypesForBookingReference(reference: String): List<NotificationEventType>? {
     return visitSchedulerClient.getNotificationsTypesForBookingReference(reference)
   }
+
+  private fun populateVisitorNames(visits: List<VisitDto>) {
+    val prisonersContactMap = mutableMapOf<String, List<PrisonerContactDto>>()
+    visits.forEach { visit ->
+      if (prisonersContactMap[visit.prisonerId] == null) {
+        val contacts = try {
+          getPrisonerContacts(visit.prisonerId)
+        } catch (e: NotFoundException) {
+          LOG.info("No contacts found for prisoner id - ${visit.prisonerId}")
+          emptyList()
+        }
+        prisonersContactMap[visit.prisonerId] = contacts
```

Comment:
adding contacts to a map o ensure we are not going back to contact registry for the same prisonerId